[Feature] 프로필 UI 구현

dudtjs1021ej opened the pull request:
### 📕 Issue

x

### 📙 작업 내역

- [x] 프로필, 즐겨찾기버튼, 나의코디버튼 이미지 추가
- [x] SFProDisplay 폰트 추가
- [x] 프로필 UI CollectionView로 구현
	- [x] DiffableDataSource 사용
	- [x]  프로필, 메뉴뷰를 HeaderView로 구현



### 📘 작업 유형

- [x] 신규 기능 추가
- [ ] 버그 수정
- [ ] 리펙토링
- [ ] 문서 업데이트


### 📋 체크리스트

- [x] Merge 하는 브랜치가 올바른가?
- [x] PR과 관련없는 변경사항이 없는가?
- [x] 내 코드에 대한 자기 검토가 되었는가?
- [ ] 변경사항이 효과적이거나 동작이 작동한다는 것을 보증하는 테스트를 추가하였는가?
- [ ] 새로운 테스트와 기존의 테스트가 변경사항에 대해 만족하는가?

### 📱 테스트 방법
x

### 📝 리뷰 노트

- collectionView headerView로 프로필과 메뉴를 넣었고 피드사진 cell로 구현했습니다.
- collectionView는 delegateFlowLayout, DiffableDataSource 사용했습니다.
- 지금 코드로는 profileViewController를 볼 수 없고 나중에 탭바랑 합칠 예정입니다!

### 🏞️ 스크린샷

<image src = "https://user-images.githubusercontent.com/77915491/206728983-16620a66-8d1a-40a4-9dbe-b1a79db2ca29.png" width="25%">

dudtjs1021ej committed:
✨ [feat] 프로필 피드 이미지 cell 구현

- collectionView를 사용해 프로필 피드 이미지 cell 구현
- 프로필 샘플 이미지 추가

dudtjs1021ej committed:
♻️ [refactor] collectionView DiffableDataSource로 변경

dudtjs1021ej committed:
🚚 [style] 코디버튼, 즐겨찾기버튼 이미지 추가

dudtjs1021ej committed:
🚚 [style] SFProDisplay 폰트 추가

dudtjs1021ej committed:
✨ [feat] headerView 구현

- 프로필을 보여주는 ProfileView 구현
- 나의코디, 즐겨찾기 버튼이 있는 MenuView 구현

dudtjs1021ej committed:
Merge branch 'develop' of https://github.com/YAPP-Github/21st-iOS-Team-2-iOS into feature/profile-ios2-73

Daltonicc commented on Fitfty/Projects/Profile/Sources/View/FeedImageCell.swift:
```diff
@@ -0,0 +1,40 @@
+import Common
+
+final class FeedImageCell: UICollectionViewCell {
+    static let identifier: String = "FeedImageCell"
```
[P3] 썬 요거는 identifier직접 만들지마시고 아리가 만들어둔 className 프로퍼티 활용하시면 좋을 것 같아요 !

Daltonicc commented on Fitfty/Projects/Profile/Sources/View/HeaderView.swift:
```diff
@@ -0,0 +1,66 @@
+        settingButton.widthAnchor.constraint(equalToConstant: 62).isActive = true
+        settingButton.heightAnchor.constraint(equalToConstant: 34).isActive = true
+        settingButton.topAnchor.constraint(equalTo: topAnchor, constant: 12).isActive = true
+        settingButton.rightAnchor.constraint(equalTo: rightAnchor, constant: -20).isActive = true
```
[P3] 각각의 constraint를 일일히 active 시켜주기보다는 NSLayoutConstraint의 activate 메서드를 활용해보시는게 어떨까요 ! 요게 더 효율적이라고 하네요 ! 참고) https://developer.apple.com/documentation/uikit/nslayoutconstraint/1526955-activate

Daltonicc commented on Fitfty/Projects/Profile/Sources/View/MenuView.swift:
```diff
@@ -0,0 +1,90 @@
+        bookmarkTextButton.topAnchor.constraint(equalTo: bookmarkIconButton.bottomAnchor, constant: 14).isActive = true
+    }
+    
+    @objc func didTapCodyButton(_ sender: UIButton) {
```
[P3] MenuView의 state(cody, bookmark)를 만들어서 처리해보시는게 어떨까요 ! 각 state에 대한 ui 업데이트를 함수로 분리해주시고 코디 버튼과 북마크버튼을 입력했을때 MenuView의 state만 바꿔주는거죠 요렇게 처리해두시는게 장기적으로 관리하시기 편할 것 같아요 !

Daltonicc commented on Fitfty/Projects/Profile/Sources/ViewController/ProfileViewController.swift:
```diff
@@ -0,0 +1,120 @@
+        collectionView.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
+    }
+    
+    func setUpDataSource() {
```
[P4] 함수의 네이밍과 다르게 데이터소스 외의 로직도 처리해주고 있어요 register 코드나 delegate 지정 코드는 setUpCollectionView요런 메서드로 옮기시는게 어떨까 생각합니다 !!

Daltonicc reviewed (commented):
썬 고생하셨어요!!😄👍🏿 리뷰 확인하시구 만약 수정한 내용있으면 수정한 커밋해시 해당 코멘트에 남겨놔주시면 확인할게요 !!

dudtjs1021ej committed:
Merge commit 'refs/pull/6/head' of https://github.com/YAPP-Github/21st-iOS-Team-2-iOS into feature/profile-ios2-73

dudtjs1021ej committed:
🔧 [chore] 리뷰 기반 코드 수정

- identifier className프로퍼티 사용하도록 변경
- NSLayoutConstraint의 activate 메서드 활용
- MenuView에 MenuState 추가

dudtjs1021ej commented on Fitfty/Projects/Profile/Sources/View/FeedImageCell.swift:
```diff
@@ -32,9 +31,11 @@ final class FeedImageCell: UICollectionViewCell {
-        feedImageView.bottomAnchor.constraint(equalTo: self.bottomAnchor).isActive = true
-        feedImageView.leftAnchor.constraint(equalTo: self.leftAnchor).isActive = true
-        feedImageView.rightAnchor.constraint(equalTo: self.rightAnchor).isActive = true
+        NSLayoutConstraint.activate([
+            feedImageView.topAnchor.constraint(equalTo: self.topAnchor),
+            feedImageView.bottomAnchor.constraint(equalTo: self.bottomAnchor),
+            feedImageView.leftAnchor.constraint(equalTo: self.leftAnchor),
+            feedImageView.rightAnchor.constraint(equalTo: self.rightAnchor)
+        ])
```
NSLayoutConstraint의 activate 메서드로 모두 수정했습니다😎 요게 더 효율적이라니 알려주셔서 감사합니다!

dudtjs1021ej commented on Fitfty/Projects/Profile/Sources/View/MenuView.swift:
```diff
@@ -63,28 +71,41 @@ final class MenuView: UIView {
+        ])
+    }
+    
+    private func checkMenuState() {
+        switch menuState {
+        case .cody:
+            codyIconButton.setImage(CommonAsset.Images.btnCodySelected.image, for: .normal)
+            bookmarkIconButton.setImage(CommonAsset.Images.btnBookmarkUnselected.image, for: .normal)
+        case .bookmark:
+            codyIconButton.setImage(CommonAsset.Images.btnCodyUnselected.image, for: .normal)
+            bookmarkIconButton.setImage(CommonAsset.Images.btnBookmarkSelected.image, for: .normal)
+        }
```
menuState 열거형을 만들어서 checkMenuState에서 UI 업데이트를 하게끔 구현해봤는데 제가 잘 이해한걸까요?!

Daltonicc replied:
꾿꾿 맞습니당 하나만 더 말씀드리면 버튼을 클릭할 때마다 checkMenuState메서드를 호출하지마시고 menuState에 프로퍼티 감시자를 만들어서 체크해보시는게 어떨까요 !

dudtjs1021ej replied:
[`96cdd92`](https://github.com/YAPP-Github/21st-iOS-Team-2-iOS/pull/7/commits/96cdd924070cb755f8b0a48b33861219e80e8553)
didSet 으로 UI 업데이트 하게끔 수정했습니다!
확인하시면 바로 머지할게요

dudtjs1021ej commented on Fitfty/Projects/Profile/Sources/ViewController/ProfileViewController.swift:
```diff
@@ -33,26 +33,27 @@ final public class ProfileViewController: UIViewController {
-                                withReuseIdentifier: HeaderView.identifier)
-        
+                                withReuseIdentifier: HeaderView.className)
+    }
```
className 프로퍼티로 수정 완료했습니다! (아리 감사합니당)

dudtjs1021ej committed:
🔧 [chore] ProfileViewController setUpCollectionView 메소드 추가

dudtjs1021ej reviewed (commented):
@Daltonicc 리뷰 감사합니다! 모두 수정 했습니다😆

dudtjs1021ej commented on Fitfty/Projects/Profile/Sources/ViewController/ProfileViewController.swift:
```diff
@@ -40,6 +41,9 @@ private extension ProfileViewController {
         collectionView.translatesAutoresizingMaskIntoConstraints = false
+    }
+    
+    func setUpCollectionView() {
         collectionView.delegate = self
         collectionView.register(FeedImageCell.self,
                                 forCellWithReuseIdentifier: FeedImageCell.className)
```
setUpCollectionView 메소드 추가했습니다!

dudtjs1021ej reviewed (commented):
리뷰 한개 빠트렸네요..!😅

Daltonicc commented:
앗 썬 혹시 아리가 작업한 브랜치 썬 브랜치로 땡겨놓으신건가요??

dudtjs1021ej commented:
> 앗 썬 혹시 아리가 작업한 브랜치 썬 브랜치로 땡겨놓으신건가요??

아리가 만들어놓은 className프로퍼티 쓰려고 땡겨왔는데 혹시 문제 있을까요..???

dudtjs1021ej committed:
Merge branch 'develop' of https://github.com/YAPP-Github/21st-iOS-Team-2-iOS into feature/profile-ios2-73

dudtjs1021ej committed:
♻️ [refactor] MenuView UI 업데이트 didSet 사용
